asio/include/asio/impl/error_code.ipp

```diff
@@ -98,17 +98,14 @@ public:
 #if defined(__sun) || defined(__QNX__) || defined(__SYMBIAN32__)
     using namespace std;
     return strerror(value);
-#elif defined(__MACH__) && defined(__APPLE__) \
-  || defined(__NetBSD__) || defined(__FreeBSD__) || defined(__OpenBSD__) \
-  || defined(_AIX) || defined(__hpux) || defined(__osf__) \
-  || defined(__ANDROID__)
+#elif defined(__GLIBC__) && defined(_GNU_SOURCE)
```

Comment:
Musl library is not gnu but also have strerror_r implemented.

http://git.musl-libc.org/cgit/musl/tree/include/string.h#n64

I don't think it will work with `defined(_GNU_SOURCE)` for this case.